src/test/java/com/puppycrawl/tools/checkstyle/internal/AllTestsTest.java

```diff
@@ -191,9 +193,11 @@ private static void verifyInputFile(Map<String, List<String>> allTests, boolean
             }
         }
 
-        assertTrue(found, "Resource must be named after a Test like 'InputMyCustomCase.java' "
-                + "and be in the sub-package of the test like 'mycustom' "
-                + "for test 'MyCustomCheckTest': " + path);
+        if (!skipFileNaming) {
```

Comment:
in case you skip assertion with this flag, all method logic above does not make sense, you can just do not invoke this method if flag is true.

Reply:
changed

Reply:
This condition make `checkInputMatchCorrectFileStructure(classes, folderPath, skipFileNaming` illogical as  `skipFileNaming` is used inside, but never would be `true`

Reply:
Reverted